extensions/mssql/src/dashboard/bookWidget.ts

```diff
@@ -0,0 +1,120 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import {
+	BookContributionProvider, BookContribution
+} from './bookExtensions';
+
+const localize = nls.loadMessageBundle();
+
+export function registerBooksWidget(bookContributionProvider: BookContributionProvider): void {
+	azdata.ui.registerModelViewProvider('books-widget', async (view) => {
+		const container = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '100%',
+			height: '100%',
+			alignItems: 'left'
+		}).component();
+		const bookslocationContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '270px',
+			height: '100%',
+			alignItems: 'left',
+			position: 'absolute'
+		}).component();
+
+		let books = bookContributionProvider.contributions.map(contribution => {
+			const bookRow = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'row'
+			}).component();
+			const tsgbooklink = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+				label: contribution.name, //localize('troubleshootingBooks', 'Troubleshooting Book'),
+				title: contribution.name // localize('troubleshootingBooksTitle', 'Troubleshooting Book'),
+			}).component();
+			tsgbooklink.onDidClick(() => {
+				PromptForFolder(contribution);
+			});
+			bookRow.addItem(tsgbooklink, {
+				CSSStyles: {
+					'width': '100%',
+					'color': '#0078d4',
+					'text-decoration': 'underline',
+					'padding-top': '10px',
+					'text-align': 'left'
+				}
+			});
+			return bookRow;
+		});
+
+
+		container.addItems(books, {
+			CSSStyles: {
+				'padding-left': '10px',
+				'border-top': 'solid 1px #ccc',
+				'box-sizing': 'border-box',
+				'user-select': 'text'
+			}
+		});
+		bookslocationContainer.addItem(container, {
+			CSSStyles: {
+				'padding-top': '25px',
+				'padding-left': '5px'
+			}
+		});
+		await view.initializeModel(bookslocationContainer);
+	});
+}
+
+async function PromptForFolder(bookContribution: BookContribution) {
+	let filter = {
```

Comment:
Can we wrap everything in a try/catch? Unhandled promise exceptions are not preferred, and since this is the highest level function we should cover it here.

Reply:
Sure